redpanda/cloud/cloud.go

```diff
@@ -171,7 +175,7 @@ func SpawnConn(url, authToken string) (*grpc.ClientConn, error) {
 			rl.Limiter,
 			// Retry interceptor
 			grpcretry.UnaryClientInterceptor(
-				grpcretry.WithCodes(codes.Unavailable, codes.Unknown, codes.Internal),
+				grpcretry.WithCodes(codes.Unavailable, codes.Unknown, codes.Internal, codes.Unauthenticated),
```

Comment:
Not a blocker: Could the CI uses the credentials from the secret manager that would authenticate against our cloud? 

I think it should be under `sdlc/prod/buildkite/rpk_test_client`

Reply:
I've got creds and vaults set up for secret manager that are per repo across several repos. I'd prefer to not collide with other testing/deal with potential quotas 😅